### PR TITLE
Added a _flush method to Duplex

### LINKED ISSUE #46: Add _flush to Duplex

### PR DESCRIPTION
**Pull request by oppenlander**
Fixes #46

### CODE REVIEW

File: lib/streams/compound-duplex.js [hunk 1]
```diff
@@ -63,12 +63,40 @@ function CompoundDuplex(writable, readable, options) {
 		self.push(null);
 	});
 
-	this.on('finish', function() {
-		writable.end();
+	readable.on('finish', function() {
```
Comment (crispy1989) [inline lib/streams/compound-duplex.js]: Why is this needed, and how is it a substitute for the original version?

The function of the original version was to propagate the EOF from the duplex stream (as the base class) to the next link in the pipeline (ie, the readable end of the compound duplex).

The changed version doesn't seem to do that at all; I can't tell what it's doing.  Why is it listening for 'finish' on a readable stream, when Readable doesn't even emit 'finish' events?

Comment (crispy1989) [inline lib/streams/compound-duplex.js]: I see now that you're emitting the "EOF" in _duplexPrefinish.  But still, you probably shouldn't be listening to 'finish' on a readable stream (when readables don't even emit finish) and you probably shouldn't be manually emitting 'finish' (it's an event emitted from the streams internals, and I've had problems in the past when I've tried to emit it manually ... internal stream state seems to get desync'd or something).

Comment (oppenlander) [inline lib/streams/compound-duplex.js]: Manually pushing the EOF is what also happens after `Transform._flush`.

I'll play around with it a bit more to see if I can get away with not manually emitting a `finish` event.

Comment (oppenlander) [inline lib/streams/compound-duplex.js]: Good call. I actually didn't need any of the event listeners at the end of the constructor.

**Pull request merged**